hastory-cli/src/Hastory/Cli/OptParse/Types.hs

```diff
@@ -1,56 +1,66 @@
 module Hastory.Cli.OptParse.Types where
 
-import Data.Hastory.API (HastoryClient)
+import Data.Hastory.API (Token)
 import qualified Data.Text as T
 import Path (Abs, Dir, Path)
 
 data Arguments =
-    Arguments Command
-              Flags
-    deriving (Show, Eq)
+  Arguments Command Flags
+  deriving (Show, Eq)
 
 data Instructions =
-    Instructions Dispatch
-                 Settings
-    deriving (Show)
+  Instructions Dispatch Settings
+  deriving (Show)
 
 data Command
-    = CommandGather
-    | CommandGenGatherWrapperScript
-    | CommandListRecentDirs ListRecentDirArgs
-    | CommandChangeDir Int
-    | CommandGenChangeWrapperScript
-    | CommandSuggestAlias
-    deriving (Show, Eq)
-
-newtype ListRecentDirArgs = ListRecentDirArgs
+  = CommandGather
+  | CommandGenGatherWrapperScript
+  | CommandListRecentDirs ListRecentDirArgs
+  | CommandChangeDir Int
+  | CommandGenChangeWrapperScript
+  | CommandSuggestAlias
+  deriving (Show, Eq)
+
+newtype ListRecentDirArgs =
+  ListRecentDirArgs
     { lrdArgBypassCache :: Maybe Bool
-    } deriving (Show, Eq)
+    }
+  deriving (Show, Eq)
 
-data Flags = Flags
-    { flagCacheDir      :: Maybe FilePath
+data Flags =
+  Flags
+    { flagCacheDir :: Maybe FilePath
     , flagStorageServer :: Maybe T.Text
-    , flagStorageToken  :: Maybe T.Text
-    } deriving (Show, Eq)
+    , flagStorageToken :: Maybe T.Text
+    }
+  deriving (Show, Eq)
 
 data Configuration =
-    Configuration
-    deriving (Show, Eq)
+  Configuration
+  deriving (Show, Eq)
 
 data Dispatch
-    = DispatchGather
-    | DispatchGenGatherWrapperScript
-    | DispatchListRecentDirs ListRecentDirSets
-    | DispatchChangeDir Int
-    | DispatchGenChangeWrapperScript
-    | DispatchSuggestAlias
-    deriving (Show, Eq)
-
-newtype ListRecentDirSets = ListRecentDirSets
+  = DispatchGather
+  | DispatchGenGatherWrapperScript
+  | DispatchListRecentDirs ListRecentDirSets
+  | DispatchChangeDir Int
+  | DispatchGenChangeWrapperScript
+  | DispatchSuggestAlias
+  deriving (Show, Eq)
+
+newtype ListRecentDirSets =
+  ListRecentDirSets
     { lrdSetBypassCache :: Bool
-    } deriving (Show, Eq)
+    }
+  deriving (Show, Eq)
+
+data Settings =
+  Settings
+    { setCacheDir :: Path Abs Dir
+    , remoteStorageClientInfo :: Maybe RemoteStorageClientInfo
+    }
+  deriving (Show, Eq)
 
-data Settings = Settings
-    { setCacheDir         :: Path Abs Dir
-    , remoteStorageClient :: Maybe HastoryClient
-    } deriving (Show)
+data RemoteStorageClientInfo =
+  RemoteStorageClientInfo T.Text Token
+  deriving (Show, Eq)
```

Comment:
What's the first text? Better give these a name in a record.

Reply:
I've made it a type alias now